src/runtime.js

```diff
@@ -261,9 +283,13 @@ let tornado = {
     if (Array.isArray(val)) {
       if (placeholderNode) {
         let frag = document.createDocumentFragment();
+        this.helperContext.set('len', val.length);
         for (let i = 0, item; (item = val[i]); i++) {
+          this.helperContext.set('idx', i);
           frag.appendChild(body(item));
         }
+        this.helperContext.clear('len');
```

Comment:
having to clear the values manually seems a little weird.  very global but i guess that's by design.